### PR TITLE
Add certificate test automation

### PR DESCRIPTION
**Pull request by khushboo-rancher**
Test cases for certificate functionality in rancher.
Trusted and selfsigned certificates are being tested in these test cases.

### CODE REVIEW

File: tests/validation/tests/v3_api/test_certificate.py [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+        3. validate the ingress using endpoint
+        """
+        ingress_name = random_test_name("ingress-test")
+        host = route_entry_53_1
+        path = "/name.html"
+        rule = {"host": host,
+                "paths": [{"path": path, "workloadIds": [self.workload.id],
+                           "targetPort": "80"}]}
+        tls = {"certificateId": self.certificate_valid.id, "hosts": [host]}
+        validate_workload(self.p_client, self.workload, "deployment",
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_certificate.py:115]: what is the purpose of validating the workload here?

File: tests/validation/tests/v3_api/test_certificate.py [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+               }
+        self.ingress = self.p_client.create_ingress(
+            name="{}-2".format(ingress_name), namespaceId=ns_2.id,
+            rules=[rule], tls=[tls]
+        )
+        wait_for_ingress_to_active(self.p_client, self.ingress)
+        validate_ingress_using_endpoint(
+            self.p_client, self.ingress, [self.workload_2], certcheck=True,
+            is_insecure=True
+        )
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_certificate.py:196]: Can you delete the resource created in this method at the end? You can use the fixture `remove_resource`

File: tests/validation/tests/v3_api/test_certificate.py [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+                p_client.create_namespaced_certificate(
+                    name=cert_name_1, key=rancher_private_key,
+                    certs=rancher_cert,
+                    namespaceId=ns['name']
+                )
+                p_client.create_certificate(
+                    name=cert_name_2, key=rancher_private_key,
+                    certs=rancher_cert
+                )
+                assert e.value.error.status == 403
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_certificate.py:397]: it is hard to tell is the error coming from  create_namespaced_certificate or create_certificate, or both.

File: tests/validation/tests/v3_api/test_certificate.py [hunk 4]
```diff
@@ -0,0 +1,571 @@
...
+                certs=rancher_cert
+            )
+            assert certificate_allns_valid.issuer == 'Let\'s Encrypt ' \
+                                                     'Authority X3'
+            # Delete the resources
+            p_client.delete(certificate_valid)
+            p_client.delete(certificate_allns_valid)
+
+    @if_test_rbac
+    @pytest.mark.parametrize("role", rbac_role_list)
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_certificate.py:416]: Can you add a test case that the project owner/member cannot list the namespaces certificates that are not in their project?

File: tests/validation/tests/v3_api/test_certificate.py [hunk 5]
```diff
@@ -0,0 +1,571 @@
...
+            assert len(cert_count) > 0
+            assert len(cert_count_1) > 0
+
+            # Delete the resources
+            p_client.delete(certificate_valid)
+            p_client.delete(certificate_allns_valid)
+
+    @if_test_rbac
+    @pytest.mark.parametrize("role", rbac_role_list)
+    def test_edit_certificate(self, role):
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_certificate.py:459]: Can you add a test case that the project owner/member cannot edit the namespaces certificates that are not in their project?

File: tests/validation/tests/v3_api/test_certificate.py (hunk 1 above)
Comment (khushboo-rancher) [reply tests/validation/tests/v3_api/test_certificate.py:115]: Workload is created in test setup for each test, so just making sure it is working before creating an ingress pointing to it.

File: tests/validation/tests/v3_api/test_certificate.py (hunk 2 above)
Comment (khushboo-rancher) [reply tests/validation/tests/v3_api/test_certificate.py:196]: These are getting deleted in tear off of each tests.

File: tests/validation/tests/v3_api/test_certificate.py (hunk 5 above)
Comment (khushboo-rancher) [reply tests/validation/tests/v3_api/test_certificate.py:501]: Added test cases

**Pull request merged**